### PR TITLE
fix: Update charts to use best te unit for values

### LINKED ISSUE #519: Daily Avg. Max blob gas fee shows invalid y-axis values
image

### LINKED ISSUE #520: Daily Avg. Blob Fee and Blob gas price shows several invalid zero values on the y-axis
image

### PR DESCRIPTION
**Pull request by luis-herasme**
### Description
fix: Update `DailyAvgBlobFeeChart` and `DailyAvgBlobGasPriceChart` to use best unit for values

### Related Issue
Closes #520
Closes #519

### CODE REVIEW

File: apps/web/src/components/Charts/Transaction/DailyAvgMaxBlobGasFeeChart.tsx [hunk 1]
```diff
@@ -16,20 +18,21 @@ export type DailyAvgMaxBlobGasFeeChartProps = {
...
+  const { converted, unit } = useArrayBestUnit(avgMaxBlobGasFees);
+
   const options: EChartOption<
     EChartOption.SeriesBar | EChartOption.SeriesLine
   > = {
     ...buildTimeSeriesOptions({
       dates: days,
       axisFormatters: {
-        yAxisTooltip: (value) => formatWei(value),
-        yAxisLabel: (value) => formatWei(value),
```
Comment (PabloCastellano) [inline apps/web/src/components/Charts/Transaction/DailyAvgMaxBlobGasFeeChart.tsx:26]: Is it possible to keep yAxisLabel with the units?

image

File: apps/web/src/utils/charts.ts [hunk 2]
```diff
@@ -105,3 +107,16 @@ export function buildTimeSeriesOptions({
     grid: yUnit ? YUINT_TO_GRID[yUnit] : undefined,
   };
 }
+
+export function useArrayBestUnit(arr?: number[]) {
```
Comment (PJColombo) [inline apps/web/src/utils/charts.ts:112]: Can we rename this hook to something more descriptive such as `useScaledWeiAmounts`

Comment (PJColombo) [reply apps/web/src/utils/charts.ts:112]: Let's update it to also receive `toUnit?: EtherUnit`.  If no parameter is passed then it uses `findBestUnit` to get the target unit

File: packages/eth-format/index.ts [hunk 3]
```diff
@@ -63,6 +67,23 @@ export function findBestUnit(wei: bigint): EtherUnit {
...
+} {
+  const maxVal = Math.max(...arr);
+  const unit = findBestUnit(maxVal);
+
+  return {
+    converted: arr.map((value) => convertWei(value, unit)),
+    unit,
+  };
+}
+
```
Comment (PJColombo) [inline packages/eth-format/index.ts:86]: No need to have a new function to do this.

I'd  update the `findBestUnit` function to also support array inputs of types bigint, number, or string, and perform the logic defined in lines 78-79 when it receives an array.

Additionally, I would create a `convertWeis` function that accepts an array of bigint, number, or string and applies the logic defined in line 82.

File: packages/eth-format/index.ts
Comment (PJColombo) [inline packages/eth-format/index.ts:1]: Could we also abstract the `bigint | string | number` types into a new type called `EthAmount`?

**Review by PJColombo — CHANGES_REQUESTED** [review]
Overall, great work! I believe we can do some refactoring to further improve the code.

File: packages/eth-format/index.ts
Comment (luis-herasme) [reply packages/eth-format/index.ts:1]: Done [717605b](https://github.com/Blobscan/blobscan/pull/527/commits/717605b73efb2e1613930c1d7aae9c50acc6919c)

File: apps/web/src/utils/charts.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (luis-herasme) [reply apps/web/src/utils/charts.ts]: Done [1f10cf1](https://github.com/Blobscan/blobscan/pull/527/commits/1f10cf1310398aded44f1f0cbbe2ef91026788a9)

File: apps/web/src/hooks/useScaledWeiAmounts.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export function useScaledWeiAmounts(arr?: number[], toUnit?: EtherUnit) {
+  return useMemo<ScaledWeiAmounts>(() => {
+    if (!arr) {
+      return {
+        unit: "wei",
+        scaledValues: undefined,
+      };
+    }
+
+    const unit = toUnit ? toUnit : findBestUnit(Math.max(...arr));
```
Comment (PJColombo) [inline apps/web/src/hooks/useScaledWeiAmounts.ts:20]:
```suggestion
    const unit = toUnit ?? findBestUnit(Math.max(...arr));
```

File: apps/web/src/hooks/useScaledWeiAmounts.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  unit: EtherUnit;
+  scaledValues: string[] | undefined;
+};
+
+export function useScaledWeiAmounts(arr?: number[], toUnit?: EtherUnit) {
+  return useMemo<ScaledWeiAmounts>(() => {
+    if (!arr) {
+      return {
+        unit: "wei",
+        scaledValues: undefined,
```
Comment (PJColombo) [inline apps/web/src/hooks/useScaledWeiAmounts.ts:16]:
```suggestion
```

File: apps/web/src/hooks/useScaledWeiAmounts.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { useMemo } from "react";
+
+import type { EtherUnit } from "@blobscan/eth-units";
+import { convertWei, findBestUnit } from "@blobscan/eth-units";
+
+type ScaledWeiAmounts = {
+  unit: EtherUnit;
+  scaledValues: string[] | undefined;
```
Comment (PJColombo) [inline apps/web/src/hooks/useScaledWeiAmounts.ts:8]:
```suggestion
  scaledValue?: string[];
```

File: packages/eth-format/index.ts (hunk 3 above)
Comment (luis-herasme) [reply packages/eth-format/index.ts]: > I'd update the findBestUnit function to also support array inputs of types bigint, number, or string, and find the best unit in them
https://github.com/Blobscan/blobscan/commit/1f10cf1310398aded44f1f0cbbe2ef91026788a9#diff-e63748d4ac678f6bf062ade6788391a3bdc455132dd96343eb414bf3cde1b993R20


> Additionally, I would create a convertWeis function that accepts an array of bigint, number, or string and applies the logic defined in line 82.
https://github.com/Blobscan/blobscan/commit/1f10cf1310398aded44f1f0cbbe2ef91026788a9#diff-e63748d4ac678f6bf062ade6788391a3bdc455132dd96343eb414bf3cde1b993R24

**Review by PJColombo — COMMENTED** [review]
LGTM

Can we apply this cosmetic suggestions to make the code more concise?

File: apps/web/src/hooks/useScaledWeiAmounts.ts (hunk 6 above)
Comment (luis-herasme) [reply apps/web/src/hooks/useScaledWeiAmounts.ts]: Done https://github.com/Blobscan/blobscan/pull/527/commits/bc31683a20caa5eb9f5a254c5198b3d3b363fe8c

File: apps/web/src/hooks/useScaledWeiAmounts.ts (hunk 5 above)
Comment (luis-herasme) [reply apps/web/src/hooks/useScaledWeiAmounts.ts]: Done https://github.com/Blobscan/blobscan/pull/527/commits/bc31683a20caa5eb9f5a254c5198b3d3b363fe8c

Comment (luis-herasme) [reply apps/web/src/hooks/useScaledWeiAmounts.ts]: Done https://github.com/Blobscan/blobscan/pull/527/commits/bc31683a20caa5eb9f5a254c5198b3d3b363fe8c

File: apps/web/src/hooks/useScaledWeiAmounts.ts (hunk 4 above)
Comment (luis-herasme) [reply apps/web/src/hooks/useScaledWeiAmounts.ts]: Done [bc31683](https://github.com/Blobscan/blobscan/pull/527/commits/bc31683a20caa5eb9f5a254c5198b3d3b363fe8c)

**Review by PabloCastellano — APPROVED** [review]
Can you add some screenshots before and after to the PR description?

Works great!

**Pull request merged**